collect_app/src/main/java/org/odk/collect/android/configure/SettingsImporter.java

```diff
@@ -62,7 +62,13 @@ public boolean fromJSON(@NonNull String json) {
         loadDefaults(generalSharedPrefs, generalDefaults);
         loadDefaults(adminSharedPrefs, adminDefaults);
 
-        settingsChangedHandler.onSettingsChanged();
+        for (String key: generalSharedPrefs.getAll().keySet()) {
```

Comment:
Iterating through *ALL* the keys is future proofing really but it felt weird for `SettingsImporter` to just pass one through knowing it would do the right thing.

Reply:
One weirdness about this is that the behavior will be different depending on how the form update settings are ordered relative to the server URL and credential settings in the key set. Perhaps this can be considered at the same time as #4011.